src/logging/zlog/src/zlog.c

```diff
@@ -375,81 +368,14 @@ void zlog_log(enum ZLOG_SEVERITY msg_level, const char* func, const char* fmt, .
         }
     }
 
-    if (msg_level == ZLOG_ERROR)
-    {
-        zlog_request_flush_buffer();
-    }
-}
-
-bool g_flushRequested = false;
-
-void zlog_request_flush_buffer(void)
-{
-    g_flushRequested = true;
-}
-
-// Buffer flushing thread
-// Flush the thread every ZLOG_FLUSH_INTERVAL_SEC seconds
-// or when buffer is 80% full
-// or when g_flushRequested is true
-//
-// Caller should NOT hold the lock
-static void* zlog_buffer_flush_thread(void * arg)
-{
-    struct timeval tv;
-    time_t lasttime;
-
-    gettimeofday(&tv, NULL);
-    lasttime = tv.tv_sec;
-
-    do
-    {
-        time_t curtime;
-
-        sleep(ZLOG_SLEEP_TIME_SEC);
-        gettimeofday(&tv, NULL);
-        curtime = tv.tv_sec;
-        if (g_flushRequested || ((curtime - lasttime) >= ZLOG_FLUSH_INTERVAL_SEC))
-        {
-            g_flushRequested = false;
-            zlog_flush_buffer();
-            lasttime = curtime;
-        }
-        else
-        {
-            _zlog_buffer_lock();
-            if (_zlog_buffer_count >= ZLOG_BUFFER_FLUSH_MAXLINES)
-            {
-                _zlog_flush_buffer();
-            }
-            _zlog_buffer_unlock();
-        }
-    } while (zlog_buffer_flush_thread_run);
-    return NULL;
-}
-
-void zlog_init_flush_thread(void)
-{
-    zlog_buffer_flush_thread_run = true;
-    if (pthread_create(&_zlog_flush_thread, NULL, zlog_buffer_flush_thread, NULL) == 0)
+    if (msg_level == ZLOG_ERROR || (seconds - zlog_last_flushed) >= ZLOG_FLUSH_INTERVAL_SEC)
     {
-        _is_flush_thread_initialized  = true;
-    }
-}
+        _zlog_buffer_lock();
+        _zlog_flush_buffer();
```

Comment:
zlog_flush_buffer should do the lock and unlock, no?

Reply:
line 353 where it rolls the log seems to want to hold the lock for longer so I think that's why _zlog_flush_buffer() has a comment that says to acquire the lock before calling.